src/main/java/com/google/sps/app/servlets/GetEvent.java

```diff
@@ -0,0 +1,70 @@
+package com.google.sps.app.servlets;
+
+import com.google.cloud.datastore.BaseEntity;
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.FullEntity;
+import com.google.cloud.datastore.Key;
+import com.google.cloud.datastore.KeyFactory;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.sps.model.Event;
+import com.google.sps.util.dataStoreHelper;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/get-event")
+public class GetEvent extends HttpServlet {
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String tripID = request.getParameter("tripID");
+        List<Event> Event = getEvent(tripID);
+        Gson gson = new Gson();
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(Event));
+    }
+
+    public List<Event> getEvent(String userID) {
```

Comment:
An alternative route could be returning a list of event IDs and then allowing the client side to make multiple simultaneous async get requests for each event. It could possibly load up faster that way.
Just something to think about. Definitely not something to worry about right now. What do you think?

Reply:
good idea, but maybe I'll implement it later

Reply:
Let's discuss this in the next meeting. It all depends on how the UI is.